libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c

```diff
@@ -36,18 +37,90 @@
 #include "mock_queue.h"
 #include "mock_ctr_drbg.h"
 #include "mock_entropy.h"
+#include "mock_sha256.h"
+#include "mock_pk.h"
+#include "mock_ecp.h"
+#include "mock_rsa.h"
+#include "mock_bignum.h"
+#include "mock_portable.h"
 
 /* PKCS #11 includes. */
 #include "iot_pkcs11_config.h"
 #include "iot_pkcs11.h"
+/* This mock must be included after pkcs11.h */
+#include "mock_iot_pkcs11_pal.h"
+
+/* mbedtls includes. */
+#include "oid.h"
 
 /* Test includes. */
 #include "unity.h"
 
 
-
+/* ============================  GLOBAL VARIABLES =========================== */
+/* Length parameters for importing EC private keys. */
+#define EC_PARAMS_LENGTH    10
+#define EC_D_LENGTH         32
+
+#define EC_PRIV_KEY_INITIALIZER                                                    \
+    {                                                                              \
+        { CKA_CLASS, &xPrivateKeyClass, sizeof( CK_OBJECT_CLASS ) },               \
+        { CKA_KEY_TYPE, &xPrivateKeyType, sizeof( CK_KEY_TYPE ) },                 \
+        { CKA_LABEL, pucLabel, ( CK_ULONG ) strlen( ( const char * ) pucLabel ) }, \
+        { CKA_TOKEN, &xTrue, sizeof( CK_BBOOL ) },                                 \
+        { CKA_SIGN, &xTrue, sizeof( CK_BBOOL ) },                                  \
+        { CKA_EC_PARAMS, pxEcParams, EC_PARAMS_LENGTH },                           \
+        { CKA_VALUE, pxD, EC_D_LENGTH }                                            \
+    }
+
+
+/* Length parameters for importing RSA-2048 private keys. */
+#define MODULUS_LENGTH        pkcs11RSA_2048_MODULUS_BITS / 8
+#define E_LENGTH              3
+#define D_LENGTH              pkcs11RSA_2048_MODULUS_BITS / 8
+#define PRIME_1_LENGTH        128
+#define PRIME_2_LENGTH        128
+#define EXPONENT_1_LENGTH     128
+#define EXPONENT_2_LENGTH     128
+#define COEFFICIENT_LENGTH    128
+
+#define RSA_PRIV_KEY_INITIALIZER                                                   \
+    {                                                                              \
+        { CKA_CLASS, &xPrivateKeyClass, sizeof( CK_OBJECT_CLASS ) },               \
+        { CKA_KEY_TYPE, &xPrivateKeyType, sizeof( CK_KEY_TYPE ) },                 \
+        { CKA_LABEL, pucLabel, ( CK_ULONG ) strlen( ( const char * ) pucLabel ) }, \
+        { CKA_TOKEN, &xTrue, sizeof( CK_BBOOL ) },                                 \
+        { CKA_SIGN, &xTrue, sizeof( CK_BBOOL ) },                                  \
+        { CKA_MODULUS, pxRsaParams->modulus + 1, MODULUS_LENGTH },                 \
```

Comment:
This initialization is based on the assumption that the mbedTLS MPI encoding always starts with a length byte, correct? But isn't the MPI encoding stated to be opaque? I.e. it could, in theory, change.

Reply:
I aligned this code with what we are using in the provisioning code. Let me make a ticket to investigate this, I'm unsure if a refactor this is within the scope of this PR as we'll likely want to make the same change across the whole code base at once.